logstash-core/lib/logstash/pipeline.rb

```diff
@@ -696,10 +701,21 @@ def stalling_threads_info
       .each {|t| t.delete("status") }
   end
 
+  def collect_dlq_stats
+    dlq_metric = @metric.namespace([:stats, :pipelines, pipeline_id.to_s.to_sym, :dlq])
+    enabled = settings.get_value("dead_letter_queue.enable")
+    dlq_metric.gauge(:enabled, enabled)
+    if enabled
+      dlq_metric.gauge(:path, @dlq_writer.get_path.to_absolute_path)
+      dlq_metric.gauge(:segment_count, @dlq_writer.get_segment_count)
```

Comment:
Would a user want to take action or make some conclusion based on the `segment_count` value?